bob/src/cleaner.rs

```diff
@@ -29,23 +32,42 @@ impl Cleaner {
             self.soft_open_blobs,
             self.hard_open_blobs,
             self.bloom_filter_memory_limit,
+            self.index_memory_limit,
         ));
     }
 
     async fn task(
         backend: Arc<Backend>,
         t: Duration,
-        soft: usize,
-        hard: usize,
+        soft: Option<usize>,
+        hard: Option<usize>,
         bloom_filter_memory_limit: Option<usize>,
+        index_memory_limit: Option<usize>,
     ) {
         let mut interval = interval(t);
         loop {
             interval.tick().await;
-            backend.close_unneeded_active_blobs(soft, hard).await;
+            if soft.is_some() || hard.is_some() {
+                let soft = soft.unwrap_or(1);
+                let hard = hard.unwrap_or(10);
+                backend.close_unneeded_active_blobs(soft, hard).await;
+            }
             if let Some(limit) = bloom_filter_memory_limit {
                 backend.offload_old_filters(limit).await;
             }
+            if let Some(limit) = index_memory_limit {
```

Comment:
Two ways to offload index memory are better to place side by side and before bloom filters offloading